applications/PfemFluidDynamicsApplication/custom_processes/set_material_properties_from_fluid_to_rigid_nodes_process.hpp

```diff
@@ -0,0 +1,343 @@
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    SetMaterialPropertiesFromFluidToRigidNodesProcess(ModelPart& rRigidModelPart,ModelPart& rFluidModelPart)
+      : mrRigidModelPart(rRigidModelPart),
+	mrFluidModelPart(rFluidModelPart)
+    {
+
+    }
+
+    /// Destructor.
+    virtual ~SetMaterialPropertiesFromFluidToRigidNodesProcess()
+    {
+    }
+
+    void operator()()
+    {
+      Execute();
+    }
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void Execute() override
+    {
+      KRATOS_TRY
+
+	double density = 0;
+      double bulk_modulus = 0;
+      double viscosity = 0;
+      double flow_index = 1;
+      double yield_shear=0;
+      double adaptive_exponent=0;
+      double static_friction=0;
+      double dynamic_friction=0;
+      double inertial_number_zero=0;
+      double grain_diameter=0;
+      double grain_density=0;
+      double regularization_coefficient=0;
+      double infinite_friction=0;
+      double inertial_number_one=0;
+      double alpha_parameter=0;
+	
+#pragma omp parallel
+      {
+	  
+	ModelPart::ElementIterator ElemBegin;
+	ModelPart::ElementIterator ElemEnd;
+	OpenMPUtils::PartitionedIterators(mrFluidModelPart.Elements(),ElemBegin,ElemEnd);
+	for ( ModelPart::ElementIterator itElem = ElemBegin; itElem != ElemEnd; ++itElem )
+	  {
+	    ModelPart::PropertiesType &elemProperties=itElem->GetProperties();
+
+	    density = elemProperties[DENSITY];
+	    bulk_modulus = elemProperties[BULK_MODULUS];
+	    viscosity = elemProperties[DYNAMIC_VISCOSITY];
+	      
+	    if(elemProperties.Has(YIELD_SHEAR)){
+	      flow_index = elemProperties[FLOW_INDEX];
+	      yield_shear = elemProperties[YIELD_SHEAR];
+	      adaptive_exponent = elemProperties[ADAPTIVE_EXPONENT];
+	    }
+
+	    if(elemProperties.Has(STATIC_FRICTION)){
+	      
+	      static_friction = elemProperties[STATIC_FRICTION];
+	      dynamic_friction = elemProperties[DYNAMIC_FRICTION];
+	      inertial_number_zero = elemProperties[INERTIAL_NUMBER_ZERO];
+	      grain_diameter = elemProperties[GRAIN_DIAMETER];
+	      grain_density = elemProperties[GRAIN_DENSITY];
+	      
+	      if(elemProperties.Has(INERTIAL_NUMBER_ONE)){
+		inertial_number_one = elemProperties[INERTIAL_NUMBER_ONE];
+		infinite_friction = elemProperties[INFINITE_FRICTION];
+		alpha_parameter = elemProperties[ALPHA_PARAMETER];
+	      }
+	      
+	      if(elemProperties.Has(REGULARIZATION_COEFFICIENT)){
+		regularization_coefficient = elemProperties[REGULARIZATION_COEFFICIENT];
+	      }
+	      
+	    }
+	    break;
+	  }
+
+      }
+	#pragma omp parallel
+      {
+	  
+	ModelPart::NodeIterator NodeBegin;
+	ModelPart::NodeIterator NodeEnd;
+	OpenMPUtils::PartitionedIterators(mrRigidModelPart.Nodes(),NodeBegin,NodeEnd);
+	for ( ModelPart::NodeIterator iNode = NodeBegin; iNode != NodeEnd; ++iNode )
+	  {
+	    iNode->GetSolutionStepValue(BULK_MODULUS)=bulk_modulus;
```

Comment:
Just a comment, I don't want to interfere with others applications. But declaring all this variables as historical is a lot of memory usage. Maybe you can use just non-historical ones

Reply:
Additionally FastGetSOlutionStepValue is faster because omits a internal check